hail/python/hailtop/aiogoogle/client/storage_client.py

```diff
@@ -744,6 +744,14 @@ async def rmtree(self, sema: asyncio.Semaphore, url: str) -> None:
             async for entry in it:
                 await pool.call(self._remove_doesnt_exist_ok, await entry.url())
 
+    async def rmtree(self, url: str, sema: Optional[asyncio.Semaphore] = None) -> None:
```

Comment:
I don't like moving sema because it makes the interface inconsistent.  With this change, some functions take it as the first argument, but rmtree teh last.